src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs

```diff
@@ -192,6 +210,22 @@ impl ContainerManager for VirtContainerManager {
             .get(container_id)
             .ok_or_else(|| Error::ContainerNotFound(container_id.clone()))?;
         c.start(process).await.context("start")?;
+
+        // Poststart Hooks:
+        // * should be run in runtime namespace
+        // * should be run after user-specific command is executed but before start operation returns
+        // * spec details: https://github.com/opencontainers/runtime-spec/blob/main/config.md#poststart
+        let c_spec = c.spec().await;
+        let state = oci::State {
+            version: String::from("ociVersion"),
+            id: c.container_id.to_string(),
+            status: oci::ContainerState::Running,
+            pid: c.pid().await as i32,
```

Comment:
As per https://github.com/kata-containers/kata-containers/issues/5321, docker relies on the pid of the container to find out the container namespace. So please let's set state.pid to the vmm thread id as the vmm is the only process that is inside the container netns on the host.

In the meantime, it means that we need to run the prestart hooks after the vmm is launched.

Reply:
Does it mean that state.pid of all OCI hooks should be set to vmm thread/process id?

Reply:
Updated!